src/datavalues.jl

```diff
@@ -29,8 +29,11 @@ function datavaluerows(x)
     return DataValueRowIterator(datavaluetype(Tables.schema(r)), r)
 end
 
+_iteratorsize(x) = x
+_iteratorsize(::Base.HasShape{1}) = Base.HasLength()
```

Comment:
I don't get the reasoning here. Why should we treat `HasShape{2}` differently from `HasShape{1}`?

Reply:
I don’t believe this will ever kick in for anything other then ``Vector``. I’m not sure, though, this is all getting very complex...